uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/FixNotMavenizedArtifactInfo.java

```diff
@@ -0,0 +1,90 @@
+import java.util.Properties;
+import java.util.TreeMap;
+
+import org.guvnor.common.services.project.model.GAV;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.guvnor.m2repo.backend.server.M2ServletContextListener.ARTIFACT_ID;
+import static org.guvnor.m2repo.backend.server.M2ServletContextListener.GROUP_ID;
+import static org.guvnor.m2repo.backend.server.M2ServletContextListener.VERSION;
+
+/**
+ * This class aim to provide maven meta information for artifacts (.jar) that don't
+ * have maven metadata built-in.
+ * <p>
+ * Maven auto-generate a `pom.properties` to store GAV information, and it also
+ * embeds the original pom file into the produced artifacts. However, this meta information
+ * is not always available if the artifact was created using a different system like ant.
+ * Even if the artifact doesn't have the necessary embedded information, those artifacts can
+ * be available in maven repositories, but the metadata are provided differently.
+ * <p>
+ * To build the GAV information out of a .jar file, this class uses as input a
+ * `not-mavenized-artifacts.properties` from classpath that provides the feedback needed
+ * to produce the required metadata. In case of an artifact is not listed in the input file,
+ * this class won't be able to provide the GAV.
+ */
+public class FixNotMavenizedArtifactInfo {
+
+    private static final Logger logger = LoggerFactory.getLogger(FixNotMavenizedArtifactInfo.class);
+
+    private final TreeMap<String, String[]> notMavenizedArtifacts = new TreeMap<>((o1, o2) -> {
+        if (o1.length() != o2.length()) {
+            return o2.length() - o1.length();
+        }
+        return o1.compareTo(o2);
+    });
+    private static final String POM_TEMPLATE = "<project xmlns=\"http://maven.apache.org/POM/4.0.0\"\n" +
+            "         xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\n" +
+            "         xsi:schemaLocation=\"http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd\">\n" +
+            "\n" +
+            "  <modelVersion>4.0.0</modelVersion>\n" +
+            "  <groupId>{groupId}</groupId>\n" +
+            "  <artifactId>{artifactId}</artifactId>\n" +
+            "  <version>{version}</version>\n" +
+            "  <packaging>jar</packaging>\n" +
+            "</project>";
+
+    public FixNotMavenizedArtifactInfo() {
+        final Properties notMavenizedArtifactsProps = new Properties();
+        try {
+            final InputStream isNotMavenizedArtifacts = M2ServletContextListener.class.getResourceAsStream("/not-mavenized-artifacts.properties");
+            if (isNotMavenizedArtifacts != null) {
+                notMavenizedArtifactsProps.load(isNotMavenizedArtifacts);
+            }
+            for (Map.Entry<Object, Object> entry : notMavenizedArtifactsProps.entrySet()) {
+                notMavenizedArtifacts.put(entry.getKey().toString() + "-", entry.getValue().toString().split(":"));
```

Comment:
Having property files with entries `artifactId=groupId:artifactId` in the properties file is redundant (and potential sourse of typo-bugs). It would have been better to have `artifactId=groupId`, in which case you wouldn't have to split and could read the artifact id from property value.

Reply:
Hi @jhrcek this is not exactly the case, for example we have this entry in the real-world properties file:
hibernate-jpa=org.hibernate.javax.persistence:hibernate-jpa-2.1-api